src/main/java/org/codehaus/mojo/license/api/DefaultDependenciesTool.java

```diff
@@ -257,6 +257,23 @@
                     depMavenProject =
                         mavenProjectBuilder.buildFromRepository( artifact, remoteRepositories, localRepository, true );
                     depMavenProject.getArtifact().setScope( artifact.getScope() );
+
+                    // In case maven-metadata.xml has different artifactId, groupId or version.
+                    if ( !depMavenProject.getGroupId().equals( artifact.getGroupId() ) )
+                    {
+                        depMavenProject.setGroupId( artifact.getGroupId() );
+                        depMavenProject.getArtifact().setGroupId( artifact.getGroupId() );
+                    }
+                    if ( !depMavenProject.getArtifactId().equals( artifact.getArtifactId() ) )
+                    {
+                        depMavenProject.setArtifactId( artifact.getArtifactId() );
+                        depMavenProject.getArtifact().setArtifactId( artifact.getArtifactId() );
+                    }
+                    if ( !depMavenProject.getVersion().equals( artifact.getVersion() ) )
+                    {
+                        depMavenProject.setVersion( artifact.getVersion() );
+                        depMavenProject.getArtifact().setVersion( artifact.getVersion() );
+                    }
```

Comment:
I am not particularly happy about setting fields returned by a Maven component. It smells after side effects, like we recently fixed in https://github.com/mojohaus/license-maven-plugin/issues/145 . But let's try, maybe it will work.